### ISSUE TITLE
ivy: translating i18n attributes on <ng-template>

### ISSUE THREAD

**Issue by matijarmk** [angular/angular#35662]
# 🐞 bug report

### Affected Package

 The issue is caused by package @angular/localize....

### Is this a regression?

It works without enabling ivy, didn't work in 9.0.2.

### Description

i18n attributes are not translating on ng-template through @ContentChildren when using <code>"enableIvy": true</code>. Works without it.

#### FilterComponent 
```Typescript 
@Component({
  selector: 'app-filter',
  templateUrl: './filter.component.html'
})
export class FilterComponent implements OnInit, OnDestroy, AfterViewInit {
  @ContentChildren(FilterDropdownDirective) dropdowns: QueryList<FilterDropdownDirective>;
```
```Html
<ng-container *ngFor="let dropdown of dropdowns">
  <label>{{dropdown.label}}</label>
</ng-container>
```
#### FilterDropdownDirective 
```Typescript 
@Directive({
  selector: '[appFilterDropdown]',
  exportAs: 'appFilterDropdown'
})
export class FilterDropdownDirective extends FilterDirective {
   @Input() label = '';
}
```

#### Usage
```Html
<app-filter
    <ng-template appFilterDropdown
                 i18n-label="Alarm status|label name"
                 label="Alarm status">
    </ng-template>
</app-filter>
```

## 🌍  Your Environment

**Angular Version:**
<pre>Angular CLI: 9.0.3
Node: 13.8.0
OS: win32 x64

Angular: 9.1.0-next.1
... animations, common, compiler, compiler-cli, core, forms
... language-service, localize, platform-browser
... platform-browser-dynamic, router
Ivy Workspace: No

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.3
@angular-devkit/build-angular     0.900.3
@angular-devkit/build-optimizer   0.900.3
@angular-devkit/build-webpack     0.900.3
@angular-devkit/core              9.0.3
@angular-devkit/schematics        9.0.3
@angular/cdk                      9.1.0
@angular/cdk-experimental         9.1.0
@angular/cli                      9.0.3
@angular/http                     7.2.16
@angular/material                 9.1.0
@ngtools/webpack                  9.0.3
@schematics/angular               9.0.3
@schematics/update                0.900.3
rxjs                              6.5.4
typescript                        3.7.5
webpack                           4.41.2<code>

</code></pre>

#### With ivy 
<img width="1901" alt="Annotation 2020-02-25 120642" src="https://user-images.githubusercontent.com/7688695/75241912-5981ab00-57c7-11ea-811d-75ef58dd5e6b.png">

#### Without ivy
<img width="1903" alt="without-ivy" src="https://user-images.githubusercontent.com/7688695/75241724-f132c980-57c6-11ea-9057-d1f72a3e78dc.png">

**Comment by AndrewKushnir** [comment]
Hi @matijarmk, thanks for reporting the issue and detailed description of the problem.

I've created a [PR with proposed fix](https://github.com/angular/angular/pull/35681). PR still requires review and final implementation might change, but I'd like to ask if you could test your app with the changes from that PR and see if the issue is resolved.

To test your app with the changes from PR #35681, you can follow an instruction in [this document on how to use PR artifacts](https://github.com/angular/angular/blob/master/docs/DEVELOPER.md#archives-for-each-package). An archive with updated `@angular/compiler` package that contains changes from the PR is available [here](https://634826-24195339-gh.circle-artifacts.com/0/angular/compiler-pr35681-2e8a0b182e.tgz) (please make sure you update other `@angular` packages to `9.0.2` to avoid version conflicts). **Note**: the mentioned archive with updated `@angular/compiler` package is for **testing purposes only** and not suitable for production usage (that archive will be removed once PR is closed).

Thank you.

**Comment by matijarmk** [comment]
Hi @AndrewKushnir I tested your PR. It seems to be working as intended. 👍

**Comment by AndrewKushnir** [comment]
Hi @matijarmk, thanks for checking app behavior with the fix. We'll follow our regular review/testing procedures to land #35681. Thank you.

**Comment by AndrewKushnir** [comment]
Hi @matijarmk, the corresponding PR was merged and the fix should be available in the upcoming 9.0.5 release in a few days. Thank you.
